mne/epochs.py

```diff
@@ -1826,8 +1826,9 @@ def __init__(self, data, info, events, tmin=0, event_id=None,
         if data.shape[0] != len(events):
             raise ValueError('The number of epochs and the number of events'
                              'must match')
-        tmax = (data.shape[2] - 1) / info['sfreq'] + tmin
-        if event_id is None:  # convert to int to make typing-checks happy
+        tmax = (data.shape[-1] - 1) / info['sfreq'] + tmin
+        if event_id is None or isinstance(event_id, int):
+            # convert to int to make typing-checks happy
```

Comment:
This will still not work, potentially injecting event_ids that should not be there (e.g., if `events` has some ids that are not in `event_id`). See the alternative fix here:

https://github.com/mne-tools/mne-python/pull/2438/files